src/main/java/de/craften/ui/swingmaterial/MaterialComboBox.java

```diff
@@ -38,19 +98,23 @@ public void paint(Graphics g) {
         g2.setRenderingHint(RenderingHints.KEY_TEXT_ANTIALIASING, RenderingHints.VALUE_TEXT_ANTIALIAS_ON);
         g2.setRenderingHint(RenderingHints.KEY_ANTIALIASING, RenderingHints.VALUE_ANTIALIAS_ON);
 
-        g.setColor(Color.WHITE);
+        g.setColor(getBackground());
         g.fillRect(0, 0, getWidth(), getHeight());
 
         g.setFont(Roboto.REGULAR.deriveFont(16f));
-        g2.setColor(Color.BLACK);
+        g.setColor(getSelectedItem() == null ? new Color(getForeground().getRGB() & 0x00FFFFFF | HINT_OPACITY_MASK, true):getForeground());
```

Comment:
See [this](https://stackoverflow.com/questions/17256644/how-does-the-bitwise-and-work-in-java) on bitwise operators and [this](http://www.cafeaulait.org/course/week2/43.html) on conditional assignment.

*Technically*, it sets a color depending on if `getSelectedItem()` returns `null`. If it is, a new `Color` is created with the RGB values of `getForeground()` and a constant Alpha value (`getRGB()` returns an ARGB int value, where the first 8 bits are dropped with `& 0x00FFFFFF` and then replaced with our own via `| HINT_OPACITY_MASK`). If it isn't, the returned value of `getForeground()` is used instead.

*Practically*, it uses a translucent version of the foreground color to display the hint if there's no item selected (thus, no content to display). If there is something selected, the foreground color is used directly.

Gotta love those one-liners.